### PR TITLE
Add Debian 10, 11 and CentOS 8 packages

### PR DESCRIPTION
**Pull request by TravisEz13**
# PR Summary

Add Debian 10, 11 and CentOS 8 packages

## PR Context

https://github.com/PowerShell/PowerShell/issues/11019

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: tools/packaging/packaging.psm1 [hunk 1]
```diff
@@ -1097,10 +1132,16 @@ function Test-Distribution
         throw "$Distribution is required for a Debian based distribution."
     }
 
-    if ($Script:DebianDistributions -notcontains $Distribution)
+    if ( ($Environment.IsUbuntu -or $Environment.IsDebian) -and $Script:DebianDistributions -notcontains $Distribution)
     {
         throw "$Distribution should be one of the following: $Script:DebianDistributions"
     }
+
+    if ( ($Environment.IsRedHatFamil) -and $Script:RedHatDistributions -notcontains $Distribution)
```
Comment (iSazonov) [inline tools/packaging/packaging.psm1]:
```suggestion
    if ( ($Environment.IsRedHatFamily) -and $Script:RedHatDistributions -notcontains $Distribution)
```

File: tools/packaging/packaging.psm1 [hunk 2]
```diff
@@ -1097,10 +1134,16 @@ function Test-Distribution
         throw "$Distribution is required for a Debian based distribution."
     }
 
-    if ($Script:DebianDistributions -notcontains $Distribution)
+    if ( ($Environment.IsUbuntu -or $Environment.IsDebian) -and $Script:DebianDistributions -notcontains $Distribution)
```
Comment (adityapatwardhan) [inline tools/packaging/packaging.psm1]: Can you add `DebianFamily` to `$Environment`?

**Review by adityapatwardhan — APPROVED** [review]

**Comment by TravisEz13** [comment]
Coordinated Package build has finished the linux portion.  So that has passed.

**Comment by iSazonov** [comment]
Should we close #10632 , #10923 and #11019?

**Comment by TravisEz13** [comment]
The first two are usually done during a release due to the cost and the process has not been completed.  I closed the third.

**Review by XOLOTL83 — COMMENTED** [review]
revisado

**Pull request merged**